draft-homburg-deleg-incremental-deleg.md

```diff
@@ -0,0 +1,621 @@
+
+# Optimized implementation
+
+Support for authoritative name servers enables optimized query behavior by resolvers with reduced (simultaneous) queries.
+{{authoritative-name-server-support}} specifies how incremental deleg supporting authoritative name servers return referral responses for delegations.
+In {{behavior-with-auth-support}} we specify how resolvers can benefit from those authoritative servers.
+
+## Authoritative name server support
+
+Incremental delegations supporting authoritative name servers include the incremental delegation information (or the NSEC(3) records showing the non-existence) in the authority section of referral responses to legacy DNS queries.
+For example, querying the zone from {{dnssec-zone}} for `www.customer5.example. A`, will return the following referral response:
+
+~~~
+;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 54349
+;; flags: qr ; QUERY: 1, ANSWER: 0, AUTHORITY: 5, ADDITIONAL: 2
+;; QUESTION SECTION:
+;; www.customer5.example.       IN      A
+
+;; ANSWER SECTION:
+
+;; AUTHORITY SECTION:
+customer5.example.      3600    IN      NS      ns.customer5.example.
+customer5.example.      3600    IN      DS      ...
+customer5.example.      3600    IN      RRSIG   DS ...
+customer5._deleg.example.       3600    IN      SVCB    1 (
+                ns.customer5.example. alpn=h2,h3
+                ipv4hint=198.51.100.5 ipv6hint=2001:db8:5::1
+                dohpath=/dns-query{?dns}
+                )
+customer5._deleg.example.       3600    IN      RRSIG   SVCB ...
+
+;; ADDITIONAL SECTION:
+ns.customer5.example.   3600    IN      A       198.51.100.5
+ns.customer5.example.   3600    IN      AAAA    2001:db8:5::1
+
+;; Query time: 0 msec
+;; EDNS: version 0; flags: do ; udp: 1232
+;; SERVER: 192.0.2.53
+;; WHEN: Mon Jul  1 20:36:25 2024
+;; MSG SIZE  rcvd: 456
+~~~
+{: #deleg-response title="An incremental deleg referral response"}
+
+The referral response in {{deleg-response}} includes the signed SVCB RRset in the authority section.
+
+As another example, querying the zone from {{dnssec-zone}} for `www.customer6.example. A`, will return the following referral response:
+
+~~~
+;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 36574
+;; flags: qr ; QUERY: 1, ANSWER: 0, AUTHORITY: 9, ADDITIONAL: 2
+;; QUESTION SECTION:
+;; www.customer6.example.       IN      A
+
+;; ANSWER SECTION:
+
+;; AUTHORITY SECTION:
+customer6.example.      3600    IN      NS      ns.customer6.example.
+customer6.example.      3600    IN      DS      ...
+customer6.example.      3600    IN      RRSIG   DS ...
+customer5._deleg.example.       1234    IN      NSEC    (
+                customer7._deleg.example.  RRSIG NSEC SVCB )
+customer5._deleg.example.       1234    IN      RRSIG   NSEC ...
+example.        1234    IN      NSEC    (
+                customer5._deleg.example.  NS SOA RRSIG NSEC DNSKEY )
+example.        1234    IN      RRSIG   NSEC ...
+example.        1234    IN      SOA     ns.example. (
```

Comment:
Why does this get a SOA record when in the unsigned case there is nothing?

Reply:
I guess we could leave out the SOA. The **signed** NSEC is enough to know the (negative) TTL.